From 6da460db6bf005488e2dc44df8eead04ddb36673 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 22 Aug 2013 08:52:17 +0100
Subject: [PATCH 032/351] ASoC: wm5110: Unmask DRC2 IRQ for Ez2Control

Unmask the DRC2 signal detect IRQ on the IRQ2 domain which will allow
the DSP to receive the signal detection status.

Change-Id: Ia41db0ba38d6dd7e907de9650042d68a52c00bc6
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/wm5110.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/sound/soc/codecs/wm5110.c b/sound/soc/codecs/wm5110.c
index 331a47b..80cb0c0 100644
--- a/sound/soc/codecs/wm5110.c
+++ b/sound/soc/codecs/wm5110.c
@@ -1693,6 +1693,16 @@ static int wm5110_open(struct snd_compr_stream *stream)
 		goto out;
 	}
 
+	ret = regmap_update_bits(arizona->regmap, ARIZONA_IRQ2_STATUS_3_MASK,
+				 ARIZONA_IM_DRC2_SIG_DET_EINT2,
+				 ARIZONA_IM_DRC2_SIG_DET_EINT2);
+	if (ret != 0) {
+		dev_err(arizona->dev,
+			"Failed to unmask DRC2 IRQ for DSP: %d\n",
+			ret);
+		goto out;
+	}
+
 	wm5110->compr_info.stream = stream;
 out:
 	mutex_unlock(&wm5110->compr_info.lock);
@@ -1710,6 +1720,9 @@ static int wm5110_free(struct snd_compr_stream *stream)
 
 	irq_set_irq_wake(arizona->irq, 0);
 	arizona_free_irq(arizona, ARIZONA_IRQ_DSP_IRQ1, wm5110);
+	regmap_update_bits(arizona->regmap, ARIZONA_IRQ2_STATUS_3_MASK,
+			   ARIZONA_IM_DRC2_SIG_DET_EINT2,
+			   0);
 
 	wm5110->compr_info.stream = NULL;
 	wm5110->compr_info.total_copied = 0;
-- 
1.7.9.5

